File: backend/src/main/java/com/coing/util/LocalTimeDeserializer.java
package com.coing.util;

import java.io.IOException;
import java.time.LocalTime;
import java.time.format.DateTimeFormatter;
import java.time.format.DateTimeParseException;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;

public class LocalTimeDeserializer extends JsonDeserializer<LocalTime> {
	private static final DateTimeFormatter FORMATTER_1 = DateTimeFormatter.ofPattern("HHmmss");
	private static final DateTimeFormatter FORMATTER_2 = DateTimeFormatter.ofPattern("HH:mm:ss");

	@Override
	public LocalTime deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
		String timeText = p.getText();
		try {
			return LocalTime.parse(timeText, FORMATTER_1);
		} catch (DateTimeParseException e) {
			return LocalTime.parse(timeText, FORMATTER_2);
		}
	}
}

File: backend/src/main/java/com/coing/util/LocalTimeDeserializer.kt
package com.coing.util

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonDeserializer
import java.io.IOException
import java.time.LocalTime
import java.time.format.DateTimeFormatter
import java.time.format.DateTimeParseException

class LocalTimeDeserializer : JsonDeserializer<LocalTime>() {

	companion object {
		private val FORMATTER_1 = DateTimeFormatter.ofPattern("HHmmss")
		private val FORMATTER_2 = DateTimeFormatter.ofPattern("HH:mm:ss")
	}

	@Throws(IOException::class)
	override fun deserialize(p: JsonParser, ctxt: DeserializationContext): LocalTime {
		val timeText = p.text
		return try {
			LocalTime.parse(timeText, FORMATTER_1)
		} catch (e: DateTimeParseException) {
			LocalTime.parse(timeText, FORMATTER_2)
		}
	}
}
